### PR TITLE
add new workflow for cutting .latest branches

### PR DESCRIPTION
**Pull request by emmyoop**
resolves https://github.com/dbt-labs/dbt-core/issues/5074

### Description

This will create the new `*.latest` branch.  **There is a known missing piece here to update `dev-requirements.txt` to point to the new `*.latest` branch.  This will need to be done manually for now**

It will also bump the version on `main` to the right alpha version and cleanup all changelogs.  The result of this part will be a PR that can be reviewed.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: .github/workflows/cut-release-branch.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  run:
+    shell: bash
+
+permissions:
+  contents: write
+
+jobs:
+  cut_branch:
+    name: "Cut branch and clean up main for dbt-snowflake"
+    uses: dbt-labs/actions/.github/workflows/cut-release-branch.yaml@main
```
Comment (mikealfare) [inline .github/workflows/cut-release-branch.yml:34]: Should this be `.yml`?

File: .github/workflows/cut-release-branch.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# **what?**
+# Cuts a new `*.latest` branch
+# Also cleans up all files in `.changes/unreleased` and `.changes/previous verion on
```
Comment (mikealfare) [inline .github/workflows/cut-release-branch.yml:3]: I don't see where this cleanup happens. Perhaps this is calling a different workflow?

Comment (emmyoop) [reply .github/workflows/cut-release-branch.yml:3]: Yes, everything happens in [actions/cut-release-branch.yaml](https://github.com/dbt-labs/actions/blob/main/.github/workflows/cut-release-branch.yaml)

Comment (emmyoop) [reply .github/workflows/cut-release-branch.yml]: I updated the comment to be more clear

**Review by McKnight-42 — APPROVED** [review]

**Pull request merged**